backend/Tournament/managers.py

```diff
@@ -0,0 +1,174 @@
+                    user=tournament.host_user,
+                    display_name=validated_data['host_user_display_name'] if validated_data['host_user_display_name'] else None,
+                )
+
+                return tournament
+
+        except Exception as e:
+            raise TournamentSetupException(e)
+
+    @staticmethod
+    def setup_tournament_matches(tournament: Tournament) -> None:        
+        if tournament.state != TournamentState.LOBBY or tournament.next_match != 0:
+            raise TournamentSetupException("Tournament is not in a valid state to set up matchups.")
+        
+        tournament_players = list(tournament.players.all())
+        if len(tournament_players) != tournament.player_amount:
+            raise TournamentSetupException("The number of tournament_players does not match the expected player amount.")
+        
+        match_amount = tournament.player_amount - 1
+
+        try:
+            with transaction.atomic():
+
+                # assignement of players to initial matches are done based on the order of the TournamentPlayers
+                # if you want to set the players for the initial matches based on some other algorithm, replace this part
+                for i in range(0, len(tournament_players), 2):
+                    match = Match.objects.create(
+                        tournament=tournament,
+                        tournament_match_id=i // 2
+                    )
+                    Player.objects.create(
+                        user=tournament_players[i].user,
+                        match=match
+                    )
+                    Player.objects.create(
+                        user=tournament_players[i+1].user,
+                        match=match
+                    )               
+
+                # create the rest of the tournament matches
+                # the match winners will be assigned to them later
+                for tournament_match_id in range(len(tournament_players) // 2, match_amount):
+                    Match.objects.create(
+                        tournament=tournament,
+                        tournament_match_id=tournament_match_id
+                    )
+
+        except Exception as e:
+            raise TournamentSetupException(f"An error occurred while setting up matchups: {e}")
+
+    @staticmethod
+    def start_tournament(tournament: Tournament) -> TournamentCreationSerializer:
+        player_amount = tournament.players.count()
+        if player_amount != tournament.player_amount:
+            raise TournamentSetupException(f"tournament must have {tournament.player_amount} players to start, but currently it has {player_amount} players.")
+        TournamentSetupManager.setup_tournament_matches(tournament)
+        tournament.start_tournament()
+        tournament.save()
+
+
+class TournamentInProgressManager:
+    @staticmethod
+    def make_sure_tournament_has_not_expired(tournament: Tournament) -> None:
```

Comment:
abort_if_expired would be more descriptive name as this does something quite fatal!

Reply:
sure

Reply:
I've removed the tournament abortion from this function as it is done in the decorator that calls it. Now it just raises the exception.